cosmos/models/Workflow.py

```diff
@@ -114,8 +119,9 @@ def __init__(self, manual_instantiation=True, *args, **kwargs):
         self.dont_garbage_collect = []
 
     @property
+    @lru_cache()
```

Comment:
Hopefully caching this will prevent Cosmos from crashing trying to write to a log when /tmp is full and sqlalchemy blows up